projects/storefrontlib/src/cms-components/organization/order-approval/order-approval-list/order-approval-list.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component, OnInit } from '@angular/core';
+import {
+  B2BSearchConfig,
+  CxDatePipe,
+  EntitiesModel,
+  OrderApproval,
+  OrderApprovalService,
+  RoutingService,
+} from '@spartacus/core';
+import { Observable } from 'rxjs';
+import { filter, map, switchMap, tap } from 'rxjs/operators';
+import {
+  AbstractListingComponent,
+  ListingModel,
+} from '../../abstract-component/abstract-listing.component';
+
+@Component({
+  selector: 'cx-order-approval-list',
+  templateUrl: './order-approval-list.component.html',
+})
+export class OrderApprovalListComponent extends AbstractListingComponent
+  implements OnInit {
+  cxRoute = 'orderApprovals';
+
+  constructor(
+    protected routingService: RoutingService,
+    protected orderApprovalService: OrderApprovalService,
+    protected cxDate: CxDatePipe
+  ) {
+    super(routingService);
+  }
+
+  ngOnInit(): void {
```

Comment:
As a best practice, avoid using ngOnInit if you can avoid it. Perhaps we can initialize this.data$ in the constructor.